quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HostnameV2PropertyMappers.java

```diff
@@ -35,13 +40,50 @@ public static PropertyMapper<?>[] getHostnamePropertyMappers() {
         .map(b -> b.isEnabled(() -> Profile.isFeatureEnabled(Profile.Feature.HOSTNAME_V2), "hostname:v2 feature is enabled").build())
         .toArray(s -> new PropertyMapper<?>[s]);
     }
-    
+
     public static void validateConfig() {
         List<String> inUse = REMOVED_OPTIONS.stream().filter(s -> Configuration.getOptionalKcValue(s).isPresent()).toList();
-        
+
         if (!inUse.isEmpty()) {
             LOGGER.errorf("Hostname v1 options %s are still in use, please review your configuration", inUse);
         }
+
+        String host = Configuration.getConfigValue(HttpOptions.HTTP_HOST).getValue();
+        String proxyHeaders = Configuration.getConfigValue(ProxyOptions.PROXY_HEADERS).getValue();
+        if (host == null) {
+            // hostname-strict=false, generally proxy-headers should be set, but it's not required
+            return;
+        }
+        if (proxyHeaders != null) {
+            // must not be tls passthrough, but there's no way of knowing the intention
+            return;
+        }
+        if (HttpPropertyMappers.isHttpsEnabled()) {
+            // must be tls passthrough, but there's no way of knowing the intention
+            return;
+        }
+        // if https is not enabled, and any part of the host does not match the rest of the system, then there's likely a problem
+        try {
+            URL url = new URL(host);
+            if ("https".equalsIgnoreCase(url.getProtocol())) {
```

Comment:
What if someone is using edge TLS termination but doesn't use proxy headers, and sets hardcoded hostname incl. the protocol (`https://`)? This would trigger the warn/error, which I believe it should not.

Reply:
> This would trigger the warn/error, which I believe it should not.

It absolutely should - that's the main thrust of this issue. We are consistently getting discusssions / issues precisely because people are not setting the proxy-headers in this case. I guess the alternative would be to have secure context detection consider the server's configured host, instead of just what is coming from the request.

Reply:
Could you elaborate why it should trigger error when the `hostname` is specified as a full URL? In that case we don't need proxy headers.

Reply:
Oh right, sorry I wasn't thinking clearly - simplified this to just a specific case.